pandas/_libs/src/ujson/python/date_conversions.c

```diff
@@ -67,7 +67,7 @@ npy_datetime NpyDateTimeToEpoch(npy_datetime dt, NPY_DATETIMEUNIT base) {
 }
 
 /* Convert PyDatetime To ISO C-string. mutates len */
-char *PyDateTimeToIso(PyDateTime_Date *obj, NPY_DATETIMEUNIT base,
+char *PyDateTimeToIso(PyObject *obj, NPY_DATETIMEUNIT base,
```

Comment:
Apparently `PyDateTime_Date` is not considered part of the public API and attempts have been made in the past to privatize:

https://github.com/python/cpython/pull/10238#issuecomment-434279039

So PyObject seems the right way to go here and gets rid of warnings